sysl2/sysl/fmt_grammar/FmtParser.g4

```diff
@@ -0,0 +1,60 @@
+grammar FmtParser;
+
+expression
+    : text* ((simp_exp
+                 | bool_exp
+                 | cmp_exp
+                 | text) text*)*
+    ;
+yes_stmt
+    : text* ((simp_exp
+                 | bool_exp
+                 | cmp_exp
+                 | text) text*)*
+    ;
+no_stmt
+    : text* ((simp_exp
+                 | bool_exp
+                 | cmp_exp
+                 | text) text*)*
+    ;
+cmp_exp_body
+    : param eq value QM yes_stmt (BAR no_stmt)?
+    ;
+cmp_exp
+    : EXP_OPEN cmp_exp_body EXP_CLOSE
+    ;
+pattern_exp_body
+    :param PATTERN value QM yes_stmt (BAR no_stmt)?
+    ;
+bool_exp_body
+    :param QM yes_stmt (BAR no_stmt)?
+    ;
+bool_exp
+    : EXP_OPEN (bool_exp_body|pattern_exp_body) EXP_CLOSE
+    ;
+simp_exp
+    : EXP_OPEN param EXP_CLOSE
+    ;
+
+text
+    : TSTR | STRING | QSTRING
+    ;
+param
+    : TSTR | STRING | QSTRING
+    ;
```

Comment:
again, what is the difference between `text` and `param`

Reply:
they are overlapping, I need to know it's a param or just a normal text.